### PR TITLE
ENH: adonis: validate metadata to check for NaNs

### LINKED ISSUE #265: ENH: adonis: validate metadata to check for NaNs
**Improvement Description**
Generate a user-friendly error when NaNs are present in the metadata

**Current Behavior**
R spews some nonsense about non-conformable arrays

**Proposed Behavior**
Catch this with some simple metadata validation, or capture the error message and report something more user-friendly.

### PR DESCRIPTION
**Pull request by collin-le**
Fixes #265

### CODE REVIEW

File: q2_diversity/_beta/_visualizer.py [hunk 1]
```diff
@@ -365,6 +365,12 @@ def adonis(output_dir: str,
         for i in t.factors:
             metadata.get_column(i.name())
 
+    # Check metadata for NaNs
+    metadata = metadata.to_dataframe()
+    if metadata.isnull().values.any() == True:
+        raise ValueError("Adonis cannot run with NaNs in metadata.")
+    metadata = qiime2.Metadata(metadata)
```
Comment (thermokarst) [inline q2_diversity/_beta/_visualizer.py:372]:
```suggestion
    metadata_df = metadata.to_dataframe()
    if metadata_df.isnull().values.any():
        raise ValueError("Adonis cannot run with NaNs in metadata.")
```

Comment (collin-le) [reply q2_diversity/_beta/_visualizer.py:372]: Yes, your suggested changes look great! I noticed the adonis test file, but was not sure if it was necessary/what would need to be done for that. Can you please explain what needs to be done to create the test?

Comment (thermokarst) [reply q2_diversity/_beta/_visualizer.py:372]: Thanks @collin-le!

I think in the adonis test suite we could just copy a test like this:

https://github.com/qiime2/q2-diversity/blob/406fedf67a574ecd8885f63dc58e06bd3ff5147a/q2_diversity/tests/test_adonis.py#L78-L84

and change the test data to be something with NaNs in the metadata. Then, the `self.assertRaisesRegex` part could match on some part of your error message above (`"Adonis cannot run with NaNs in metadata."`).

**Comment by thermokarst** [comment]
@collin-le   - do you think you can add a short entry for this change into the release changelog?

https://forum.qiime2.org/t/preview-qiime-2-2021-2-development-changelog/17815

**Pull request merged**